### PR TITLE
Don't convert implicitly non-str arguments to istr

### PR DESCRIPTION
**Pull request by gsakkis**
Fix `istr` methods to not convert implicitly non-str arguments to istr.

Before this PR:
```py
In [1]: from istr import istr

In [2]: istr("hey") > 5
Out[2]: True
```

After this PR:
```py

In [1]: from istr import istr

In [2]: str("hey") > 5
TypeError: '>' not supported between instances of 'str' and 'int'
```

### CODE REVIEW

**Comment by dmurphy18** [comment]
@gsakkis All code changes require tests (written with Pytest) which exercises the code changes before they can be accepted. Will be more than happy to review once there are tests ensuring the code changes are working correctly.

**Comment by gsakkis** [comment]
@dmurphy18 I updated the PR with added tests.

**Comment by gsakkis** [comment]
@dmurphy18 could you take a look?

**Comment by dmurphy18** [comment]
@gsakkis In future, just add persons to Reviewers, github should email them to review then.
Looking at it now

**Comment by dmurphy18** [comment]
getting second opinion from @s0undt3ch

**Comment by s0undt3ch** [comment]
Need review from @msteed

File: tests/test_match.py
Comment (Jamim) [inline tests/test_match.py:158]: Hello @gsakkis,
I don't think this loop should be in this particular test since:
  * it performs exactly the same comparison as for `test_ne_different`, so it won't increase the coverage
  * it checks completely opposite to what a test name means
```suggestion
```

**Review by Jamim — CHANGES_REQUESTED** [review]

File: tests/test_match.py
Comment (gsakkis) [reply tests/test_match.py]: Thanks @Jamim, i removed the redundant checks.

**Review by msteed — APPROVED** [review]

**Comment by msteed** [comment]
Looks good to me. I don't have write access to this repo so someone else will have to merge.

**Review by Jamim — APPROVED** [review]

**Review by dmurphy18 — APPROVED** [review]

**Comment by gsakkis** [comment]
Thanks all! Any idea of when there will be a new PyPI release?

**Pull request merged**
